OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  PushNotificationManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+import Swinject
+
+public protocol PushNotificationsProvider {
+    func registerWithDeviceToken(deviceToken: Data)
+    func didFailToRegisterForRemoteNotificationsWithError(error: Error)
+}
+
+protocol PushNotificationsListener {
+    func shouldListenNotification(userinfo: [AnyHashable: Any]) -> Bool
+    func didReceiveRemoteNotification(userInfo: [AnyHashable: Any])
+}
+
+extension PushNotificationsListener {
+    func didReceiveRemoteNotification(userInfo: [AnyHashable: Any]) {
+       guard let dictionary = userInfo as? [String: Any], shouldListenNotification(userinfo: userInfo) else { return }
+       let link = PushLink(dictionary: dictionary)
+       if let deepLinkManager = Container.shared.resolve(DeepLinkManager.self) {
+           deepLinkManager.processNotification(with: link)
+       }
+   }
+}
+
+class PushNotificationsManager {
+    private var providers: [PushNotificationsProvider] = []
+    private var listeners: [PushNotificationsListener] = []
+    
+    // Init manager
+    public init(config: ConfigProtocol) {
+        self.providers = self.providersFor(config: config)
+        self.listeners = self.listenersFor(config: config)
+    }
+    
+    private func providersFor(config: ConfigProtocol) -> [PushNotificationsProvider] {
+        var rProviders: [PushNotificationsProvider] = []
```

Comment:
How about renaming it to providers, pushProviders or pProviders? as rProviders doesn't go with the feature.